Realm/Realm/Sync/Session.cs

```diff
@@ -185,5 +225,34 @@ internal void CloseHandle(bool waitForShutdown = false)
         }
 
         internal void ReportErrorForTesting(int errorCode, string errorMessage, bool isFatal) => Handle.ReportErrorForTesting(errorCode, errorMessage, isFatal);
+
+        internal static void RaiseError(Session session, Exception error)
+        {
+            var args = new ErrorEventArgs(error);
+            Error?.Invoke(session, args);
+        }
+
+        private void SubscribeNotifications()
+        {
+            var cb = new Action<string>((propertyName) => NotifyPropertyChanged(propertyName));
+            var callbackHandle = GCHandle.Alloc(cb, GCHandleType.Weak);
+            var nativeTokens = Handle.RegisterPropertyChangedCallbacks(GCHandle.ToIntPtr(callbackHandle));
+
+            _propertyChangedNotificationTokens = NotificationToken.Create(nativeTokens, (tokens) =>
+            {
+                Handle.UnregisterPropertyChangedCallbacks(tokens);
+            });
+        }
+
+        private void UnsubscribeNotifications()
+        {
+            _propertyChangedNotificationTokens?.Dispose();
+            _propertyChangedNotificationTokens = null;
+        }
+
+        private void NotifyPropertyChanged([CallerMemberName] string propertyName = "")
```

Comment:
Let's make this internal and call it `RaisePropertyChanged` to be consistent with the naming in `RealmObjectBase`.